src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -1285,31 +1324,26 @@ BuildCachedShardList(CitusTableCacheEntry *cacheEntry)
 		ErrorIfInconsistentShardIntervals(cacheEntry);
 	}
 
-	/*
-	 * We set these here, so ResetCitusTableCacheEntry() can see what has been
-	 * entered into DistShardCacheHash even if the following loop is interrupted
-	 * by throwing errors, etc.
-	 */
 	cacheEntry->sortedShardIntervalArray = sortedShardIntervalArray;
 	cacheEntry->shardIntervalArrayLength = 0;
 
 	/* maintain shardId->(table,ShardInterval) cache */
 	for (int shardIndex = 0; shardIndex < shardIntervalArrayLength; shardIndex++)
 	{
 		ShardInterval *shardInterval = sortedShardIntervalArray[shardIndex];
-		bool foundInCache = false;
+		int64 shardId = shardInterval->shardId;
 		int placementOffset = 0;
+		bool foundShardIdInCache = false;
 
-		ShardCacheEntry *shardEntry = hash_search(DistShardCacheHash,
-												  &shardInterval->shardId, HASH_ENTER,
-												  &foundInCache);
-		if (foundInCache)
-		{
-			ereport(ERROR, (errmsg("cached metadata for shard " UINT64_FORMAT
-								   " is inconsistent",
-								   shardInterval->shardId),
-							errhint("Reconnect and try again.")));
-		}
```

Comment:
Should this check really be removed?

Reply:
Reading this better it might be okay since we do `HASH_ENTER` and not `HASH_FIND`.

Reply:
I just confirmed that this is change is on purpose. When adding this check back very basic operations started failing. I changed the code slightly to not have this `foundShardIdInCache` variable at all, by passing in `NULL` as the last argument.